ui/ducks/bridge/actions.ts

```diff
@@ -86,3 +118,330 @@ export const updateQuoteRequestParams = (params: Partial<QuoteRequest>) => {
     );
   };
 };
+
+export const getBridgeERC20Allowance = async (
+  contractAddress: string,
+  chainId: Hex,
+): Promise<string> => {
+  return await submitRequestToBackground(
+    BridgeBackgroundAction.GET_BRIDGE_ERC20_ALLOWANCE,
+    [contractAddress, chainId],
+  );
+};
+
+export const submitBridgeTransaction = (
+  quoteResponse: QuoteResponse,
+  history: ReturnType<typeof useHistory>,
+) => {
+  return async (
+    dispatch: MetaMaskReduxDispatch,
+    getState: () => MetaMaskReduxState & BridgeAppState,
+  ) => {
+    const state = getState();
+    // Track event TODO
+
+    const calcFeePerGas = () => {
+      let maxFeePerGas: undefined | string;
+      let maxPriorityFeePerGas: undefined | string;
+
+      const networkAndAccountSupports1559 =
+        checkNetworkAndAccountSupports1559(state);
+
+      if (networkAndAccountSupports1559) {
+        const gasFeeEstimates = getGasFeeEstimates(state);
```

Comment:
If we want to retrieve the gas fee estimates in the UI, then we need to be using the `estimateGasFee` method from the `TransactionController` as it encapsulates alternate gas fee flows such as Linea specific gas fee estimation.

Reply:
Done in [3172b9b](https://github.com/MetaMask/metamask-extension/pull/27262/commits/3172b9bd770f25dda3fe2390852bd3b20179164d)